[quant][graphmode][fx] Add support for instance_norm

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43439 [quant][graphmode][fx] Add support for general value ops
* #43438 [quant][graphmode][fx] Add support for general shape ops
* #43437 [quant][graphmode][fx] Add support for clamp
* **#43377 [quant][graphmode][fx] Add support for instance_norm**

Summary:
Porting op tests from test_quantize_jit.py

Test Plan:
TestQuantizeFxOps

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23257045](https://our.internmc.facebook.com/intern/diff/D23257045)

jerryzh168 committed:
[quant][graphmode][fx] Add support for instance_norm

Summary:
Porting op tests from test_quantize_jit.py

Test Plan:
TestQuantizeFxOps

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

z-a-f commented on test/quantization/test_quantize_fx.py:
```diff
@@ -462,3 +465,26 @@ def test_layer_norm(self):
+        data_2d = (torch.rand((1, 4, 5, 1), dtype=torch.float),)
+        data_3d = (torch.rand((1, 4, 5, 1, 1), dtype=torch.float),)
+        data_dict = {1 : data_1d, 2 : data_2d, 3 : data_3d}
+        instance_norm_modules = {1 : nn.InstanceNorm1d,
+                                 2 : nn.InstanceNorm2d,
+                                 3 : nn.InstanceNorm3d}
+        quantized_instance_norm_modules = {
+            1 : nnq.InstanceNorm1d,
+            2 : nnq.InstanceNorm2d,
+            3 : nnq.InstanceNorm3d
+        }
+        instance_norm_ops = {1 : F.instance_norm,
+                             2 : F.instance_norm,
+                             3 : F.instance_norm}
```
nit: `Dict[int, object]` with `int` being sequential is effectively the same as a `List[object]`

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for instance_norm"


Summary:
Porting op tests from test_quantize_jit.py

Test Plan:
TestQuantizeFxOps

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23257045](https://our.internmc.facebook.com/intern/diff/D23257045)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Add support for instance_norm"


Summary:
Porting op tests from test_quantize_jit.py

Test Plan:
TestQuantizeFxOps

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23257045](https://our.internmc.facebook.com/intern/diff/D23257045)

[ghstack-poisoned]
